common/flatpak-instance.c

```diff
@@ -313,6 +317,47 @@ flatpak_instance_get_info (FlatpakInstance *self)
   return priv->info;
 }
 
+GStrv
+flatpak_instance_get_run_environ (FlatpakInstance *self)
+{
+  FlatpakInstancePrivate *priv = flatpak_instance_get_instance_private (self);
+
+  return priv->run_environ;
+}
+
+static GStrv
+get_run_environ (const char *dir)
```

Comment:
This could maybe benefit from some refactoring to share code with `flatpak_context_parse_env_block` or even `flatpak_context_parse_env_fd`.

Perhaps a common `flatpak_parse_env_block`, which returns a `GPtrArray` or `GStrv` in which every item is guaranteed to be of the form `name=value`, and takes an argument to set whether invalid variables raise an error or are quietly filtered out?

`flatpak_context_parse_env_block` and `flatpak_context_parse_env_fd` have unit tests. If this stays separate, I'd like to see unit tests for this, too, which should check for corner cases like empty variable names or missing `=`.